cmd/agent/app/reporter/grpc/reporter_test.go

```diff
@@ -71,7 +72,45 @@ func TestReporter_EmitZipkinBatch(t *testing.T) {
 		{in: &zipkincore.Span{Name: "jonatan", TraceID: 1, ID: 2, Timestamp: &a, Annotations: []*zipkincore.Annotation{{Value: zipkincore.CLIENT_SEND, Host: &zipkincore.Endpoint{ServiceName: "spring"}}}},
 			expected: model.Batch{
 				Spans: []*model.Span{{TraceID: model.NewTraceID(0, 1), SpanID: model.NewSpanID(2), OperationName: "jonatan", Duration: time.Microsecond * 1,
-					Tags: model.KeyValues{{Key: "span.kind", VStr: "client", VType: model.StringType}}, Process: &model.Process{ServiceName: "spring"}, StartTime: tm.UTC()}}}},
+					Tags:    model.KeyValues{{Key: "span.kind", VStr: "client", VType: model.StringType}},
+					Process: &model.Process{ServiceName: "spring"}, StartTime: tm.UTC()}}}},
+	}
+	for _, test := range tests {
+		err = rep.EmitZipkinBatch([]*zipkincore.Span{test.in})
+		if test.err != "" {
+			assert.EqualError(t, err, test.err)
+		} else {
+			assert.Equal(t, 1, len(handler.requests))
+			assert.Equal(t, test.expected, handler.requests[0].GetBatch())
+		}
+	}
+}
+
+func TestReporter_EmitZipkinBatch_WithAgentTags(t *testing.T) {
```

Comment:
A lot of repetition. I would suggest testing the function that merges agent tags into spans/process directly rather than going through client/server setup as in the test above.

Reply:
My bad, fixed.